### PR TITLE
MAINT: Radon (docstring, API, PEP8)

### LINKED ISSUE #744: Radon transform: Circle mode as the default
A while ago, we introduced an optional "circle mode". The reasons for this is explained here: #567.

I propose to make this circle mode the default, for the reasons listed in the referenced PR, plus the bonus of consistency with other reconstruction algorithms implemented in skimage. As this is an API change, the code should issue a FutureWarning for at least one release. Since I'm hearing rumours that a new release is approaching, now would be a good time to start issuing the warning.

### PR DESCRIPTION
**Pull request by sciunto**
Closes #744

* [X] Group references in example
* [X] move references to docstring
* [X] PEP8
* [x] new API #744 (Anyway, I'm waiting for #2228 to avoid conflict in TODO list)

Once merged, look at #1474

### CODE REVIEW

**Comment by sciunto** [comment]
@scikit-image/core This PR tackles a ticket and brings some maintenances to radon. Ready for reviews! :)

File: doc/examples/transform/plot_radon_transform.py [hunk 1]
```diff
@@ -31,9 +31,14 @@
 
 For further information on tomographic reconstruction, see
 
-- AC Kak, M Slaney, "Principles of Computerized Tomographic Imaging",
-  http://www.slaney.org/pct/pct-toc.html
-- http://en.wikipedia.org/wiki/Radon_transform
+.. [1] AC Kak, M Slaney, "Principles of Computerized Tomographic Imaging", IEEE Press 1988. http://www.slaney.org/pct/pct-toc.html
```
Comment (soupault) [inline doc/examples/transform/plot_radon_transform.py]: Wrap at 80, here and below.

File: skimage/transform/radon_transform.py [hunk 2]
```diff
@@ -23,10 +9,10 @@
 from warnings import warn
 
 
-__all__ = ["radon", "iradon", "iradon_sart"]
+__all__ = ['radon', 'order_angles_golden_ratio', 'iradon', 'iradon_sart']
 
 
-def radon(image, theta=None, circle=False):
+def radon(image, theta=None, circle=None):
```
Comment (soupault) [inline skimage/transform/radon_transform.py]: Probably should raise a deprecation warning if `circle==None`. Same for `iradon`.

File: skimage/transform/radon_transform.py [hunk 3]
```diff
@@ -404,7 +417,7 @@ def iradon_sart(radon_image, theta=None, image=None, projection_shifts=None,
         raise ValueError('Shape of projection_shifts (%s) does not match the '
                          'number of projections (%d)'
                          % (projection_shifts.shape, radon_image.shape[1]))
-    if not clip is None:
+    if clip is not None:
```
Comment (soupault) [inline skimage/transform/radon_transform.py]: Like... REALLY? Thanks for spotting this!

File: skimage/transform/radon_transform.py [hunk 4]
```diff
@@ -23,10 +9,10 @@
 from warnings import warn
 
 
-__all__ = ["radon", "iradon", "iradon_sart"]
+__all__ = ['radon', 'order_angles_golden_ratio', 'iradon', 'iradon_sart']
```
Comment (soupault) [inline skimage/transform/radon_transform.py]: I don't think that we should expose `order_angles_golden_ratio`. Not really a user-level function.

**Comment by soupault** [comment]
@sciunto easy review, thanks! :wink:.
Once addressed, LGTM.

File: skimage/transform/radon_transform.py (hunk 4 above)
Comment (sciunto) [inline skimage/transform/radon_transform.py]: I'm not sure... Indeed, it's not used internally. To me, the user has the choice to use it or not in his/her workflow.

**Comment by sciunto** [comment]
I fixed the rest. Too lazy to run the tests on my machine, I'm waiting for travis.

**Comment by sciunto** [comment]
Should be good.

File: skimage/transform/radon_transform.py [hunk 5]
```diff
@@ -50,11 +37,28 @@ def radon(image, theta=None, circle=False):
...
+    (http://www.clear.rice.edu/elec431/projects96/DSP/bpanalysis.html)
+
     """
     if image.ndim != 2:
         raise ValueError('The input image must be 2-D')
     if theta is None:
         theta = np.arange(180)
+    if circle is None:
+        warn('The default of `circle` in `skimage.transform.radon` '
+             'will change to `False` in version 0.15.')
```
Comment (jni) [inline skimage/transform/radon_transform.py]: Erm, `True`. =)

File: skimage/transform/radon_transform.py [hunk 6]
```diff
@@ -188,6 +201,10 @@ def iradon(radon_image, theta=None, output_size=None,
         else:
             output_size = int(np.floor(np.sqrt((radon_image.shape[0]) ** 2
                                                / 2.0)))
+    if circle is None:
+        warn('The default of `circle` in `skimage.transform.iradon` '
+             'will change to `False` in version 0.15.')
```
Comment (jni) [inline skimage/transform/radon_transform.py]: Also `True`!

File: skimage/transform/radon_transform.py (hunk 4 above)
Comment (jni) [inline skimage/transform/radon_transform.py]: Having looked at the docstring, to my admittedly-naive eyes, I agree that it's worthwhile exposing it.

**Comment by sciunto** [comment]
@jni I fixed the typo. Thanks for catching it.

**Comment by ahojnnes** [comment]
Thanks @sciunto!

**Comment by nnn144** [comment]
Can someone tell what does this error mean?
"serWarning: Radon transform: image must be zero outside the reconstruction circle"

**Pull request merged**